internal/workflow.go

```diff
@@ -66,8 +66,16 @@ var (
 )
 
 type (
+	// commonChannel contains methods common to both SendChannel and ReceiveChannel
+	commonChannel interface {
+		// Name returns the name of the channel.
```

Comment:
Can we add a few things to the docs: 
1. Specify that for a `SignalChannel` , `Name` is the signal name. Some places in Temporal refer to signal "Name" as "Type" so it may not obvious.
2. Specify that channels created without an explicit name have a name generated by the SDK and they should not assume it is deterministic. Normally all workflow APIs are deterministic, but in this case the channel name would not be.

Reply:
Added clarity to the documentation as requested